### PR TITLE
Fix for #665

### PR DESCRIPTION
**Pull request by ili**
Possibility to configure `CommandBehavior` for different `DataProviders`

### CODE REVIEW

File: Source/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs [hunk 1]
```diff
@@ -64,24 +64,33 @@ protected PostgreSQLDataProvider(string name, MappingSchema mappingSchema, Postg
...
+			_npgsqlTimeStampTZ    = npgSql.GetType("NpgsqlTypes.NpgsqlTimeStampTZ", false);
+			_npgsqlDate           = npgSql.GetType("NpgsqlTypes.NpgsqlDate",        true);
+			_npgsqlDateTime       = npgSql.GetType("NpgsqlTypes.NpgsqlDateTime",    false);
+			NpgsqlMacAddressType  = npgSql.GetType("NpgsqlTypes.NpgsqlMacAddress",  false);
+			NpgsqlCircleType      = npgSql.GetType("NpgsqlTypes.NpgsqlCircle",      true);
+			NpgsqlPolygonType     = npgSql.GetType("NpgsqlTypes.NpgsqlPolygon",     true);
+
+			if (npgSql.GetName().Version >= new Version(3, 2, 2))
+			{
+				_commandBehavior = CommandBehavior.KeyInfo;
```
Comment (MaceWindu) [inline Source/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs]: Just to confirm - this functionality depends only on provider version? Not on PostgreSQL version?

Comment (ili) [inline Source/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs]: As described in #665 - yes (shame on me, I just believed)

Comment (MaceWindu) [inline Source/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs]: Looks like it was added in 3.1.9 https://github.com/npgsql/npgsql/issues/1346

Comment (ili) [inline Source/DataProvider/PostgreSQL/PostgreSQLDataProvider.cs]: Great! I'v changed to 3.1.9

File: Tests/Linq/Linq/SqlExtensionTests.cs [hunk 2]
```diff
@@ -262,15 +289,15 @@ public void Build(Sql.ISqExtensionBuilder builder)
 
 
 		[Sql.Extension(               "DatePart({part}, {date})",                 ServerSideOnly = false, BuilderType = typeof(DatePartBuilder))]
-		[Sql.Extension(PN.DB2,        "",                                         ServerSideOnly = false, BuilderType = typeof(DatePartBuilderOracleInformixDb2))] // TODO: Not checked
-		[Sql.Extension(PN.Informix,   "",                                         ServerSideOnly = false, BuilderType = typeof(DatePartBuilderOracleInformixDb2))] // TODO: Not checked
+		[Sql.Extension(PN.DB2,        "",                                         ServerSideOnly = false, BuilderType = typeof(DatePartBuilderDB2))] // TODO: Not checked
+		[Sql.Extension(PN.Informix,   "",                                         ServerSideOnly = false, BuilderType = typeof(DatePartBuilderOracleInformix))] // TODO: Not checked
```
Comment (sdanyliv) [inline Tests/Linq/Linq/SqlExtensionTests.cs]: @ili, do we need these TODO's? Looks like you have checked these databases.

**Pull request merged**